[Feature] CLI: function Added plugin console/joomconsole

ThomasFinnern opened the pull request:
Plugin to support CLI commands

The joomla command line interface is a powerfule addition to jommla. It enables us to setup a joomla web page for our needs. After installation we may create users, assign usergroup and in the end JoomGallery start 'situation' 

**Actual commands:** 

It supports basic command 'joomgallery:categories:list' It contains started command  'joomgallery:categories:add'

Available commands for the "joomgallery" namespace:
*  joomgallery:category       List all variables of a joomgallery category
*  joomgallery:category:list  List all joomgallery categories
*  joomgallery:image          List all variables of a joomgallery image
*  joomgallery:image:list     List all joomgallery images

Documentation as *.md file in an other repository: [Documentation](https://github.com/ThomasFinnern/JoomGallery_fith_dev/blob/main/.jg_dev_doc/jg_4x/cli/01.JG_cli_commands.en.md) 

**Work in progress**  Help appreciated

Actually two issues stop the functionality

* JoomGallery wants to use a user which does not exist in CLI call
* create category was too complex to implement but data highway to command exists


### How to test this PR ?

for courios developers above commands are working

ThomasFinnern commented:
New version from scratch

ThomasFinnern commented on plugins/console/joomconsole/language/en-GB/index.html:
Removed

ThomasFinnern commented on administrator/com_joomgallery/src/CliCommand/Add.php:
```diff
@@ -0,0 +1,341 @@
+    $categoryModel = $this->getMVCFactory()->createModel('Category', 'Administrator');
+
+    if (!$categoryModel->save($category)) {
+//      switch ($categoryModel->getError()) {
```
Is fixed

MrMusic commented:
Thanks for this PR.
I've played with it a bit and it generally works well.

I see a problem with the following commands:
1. Execution of '`joomgallery:config:set --id=2 published 0`' does not work.
It is possible to change the value from '0' to '1', but not from '1' to '0'. It is also not possible to change the title to '0'.
It seems as if '0' is filtered out?

2. Execution of '`joomgallery:config:list`':
'Modified' is too much in the table heading (there is no 'Modified date' in the config). As a result, the output is shifted.

3. Parameters of '`joomgallery:category:list`':
There is` --owner` and there is `--created`. But 'owner' is the same as 'created_by'.
I would recommend using the column names from the database as column headings for the output if possible. Then there will be no confusion or misunderstandings.

Elfangor93 commented:
Please rebase this PR, such that is contains the commits of v4.1.0. Then the checks will succeed and the PR can be properly installed for testing. Thanks!

ThomasFinnern commented:
@MrMusic 
> 2\. Execution of '`joomgallery:config:list`':
Fixed not matching columns (not updated here)

? The idea was to have only some properties but the ones which are most interesting.
a) we may remove the command completely if it does not make much sense
b) which propeties  should be in this list
c) recommend better 'command' name  than 'list'

MrMusic commented:
>? The idea was to have only some properties but the ones which are most interesting.
a) we may remove the command completely if it does not make much sense
b) which propeties should be in this list
c) recommend better 'command' name than 'list'

I think 'list' is ok.
It is also ok that only some options are displayed with _joomgallery:config:list_.
However, the heading “_Modified_” is currently too much, as it does not exist.
![cli-config-list](https://github.com/user-attachments/assets/81062d69-832e-49c3-b849-b25369d50e65)

>The idea was that in the frontend access is by 'owner', and in category edit it is also shown as owner . So people may want to use what they know ?

Ok, you use 'owner' as a synonym for 'created_by'? Because 'created' is the date, isn't it?
I just want to avoid confusion.

ThomasFinnern commented:
Will do and finish this week

ThomasFinnern commented:
Updated and tested, 
- fixed: config set publish to zero
- fixed: joomgallery:config:list removed modified as not in table

@MrMusic It is ready to test. Please try again

MrMusic commented:
>Updated and tested, 
It is ready to test. Please try again.

Thanks! 
Maybe one last thing: Should the `@since 4.1.0` statement changed to 4.2.0?
Otherwise:
I have tested this item ✅ successfully.
Thank you very much!

ThomasFinnern commented:
Updated @since lines to "@since   4.2.0"
We may accept this now
